src/service/auth.js

```diff
@@ -0,0 +1,197 @@
+    .then(res => {
+      if (!res.success) throw new Error(res.message);
+      return res;
+    })
+    .catch(err => {
+      throw err.response.data;
+    });
+
+const authWithGitHub = code =>
+  request
+    .get(`${identityBaseUrl}/auth/github?code=${encodeURIComponent(code)}`)
+    .then(res => {
+      if (!res.success) throw new Error(res.message);
+      return res;
+    })
+    .catch(err => {
+      throw err.response.data;
+    });
+
+const logout = () => request.post(`${identityBaseUrl}/logout`);
+
+const getAuthStatus = async () => {
+  const res = {
+    status: 200,
+    expiresAt: 0,
+  };
+  try {
+    const { expiresAt } = await request.get(`${identityBaseUrl}/auth/check`);
+    res.expiresAt = expiresAt;
+  } catch (e) {
+    res.status = get(e, ['response', 'status']);
+  }
+  return res;
+};
+
+const waitLogin = () =>
+  new Promise((resolve, reject) => {
+    /* eslint-disable-next-line */
+    const handler = async function() {
+      try {
+        const { status } = await getAuthStatus();
+
+        if (status === 200 || status === 403) {
+          return resolve(status);
+        }
+
+        createTimeout(handler);
+      } catch (err) {
+        return reject(err);
+      }
+    };
+
+    return handler();
+  });
+
+const getRecoveryIdentifier = email =>
+  request
+    .get(`${identityBaseUrl}/auth/recover?email=${encodeURIComponent(email)}`)
+    .then(res => {
+      if (!res.success) throw new Error(res.message);
+
+      return res.message;
+    });
+
+const sendEmailCode = async email => {
+  const { timeout } = await request.post(`${identityBaseUrl}/auth/code`, {
+    email,
+  });
+
+  return timeout;
+};
+
+/**
+ * Send sms with code for disabling otp
+ * @param {string} email
+ * @returns {Promise<void>}
+ */
+const sendOtpRecoverSms = async email => {
+  try {
+    const res = await request.get(
+      `${identityBaseUrl}/auth/recover?email=${encodeURIComponent(email)}`,
+    );
+
+    if (!res.success) throw new Error(res.message);
+
+    return res;
+  } catch (error) {
+    const { response = {} } = error;
+    const { status } = response;
+    error.code = status;
+
+    throw error;
+  }
+};
+
+/**
+ * Disable otp setting
+ * @param {object} param
+ * @param {string} param.email
+ * @param {string} param.code
+ * @returns {Promise<void>}
+ */
+const disableOtpViaSms = async ({ email, code }) => {
+  const res = await request.post(`${identityBaseUrl}/auth/recover`, {
+    email,
+    code,
+  });
+
+  if (!res.success) throw new Error(res.message);
+
+  return res;
+};
+
+const disableOtp = (email, signature, redirectUrl) =>
+  request
+    .post(`${identityBaseUrl}/auth/recover`, {
+      email,
+      signature,
+      redirectUrl,
+    })
+    .then(res => {
+      if (!res.success) throw new Error(res.message);
+
+      return res;
+    });
+
+const getSeedTemplateUrl = () => `${identityBaseUrl}/auth/recover/template`;
```

Comment:
It can be just constant and should be defined at the top of file

Reply:
main target was split `auth` parts from `identity` for more readable.
that's why some methods not refactored and stay as is.